tests/test_persistence.py

```
@@ -0,0 +1,270 @@
+        # Create test document (temporary file)
+        with tempfile.NamedTemporaryFile(mode='w', suffix='.txt', delete=False) as doc:
+            doc.write("This is a test document about vitamins and nutrition.")
+            doc_path = Path(doc.name)
+
+        try:
+            # Create repository
+            repo = SessionRepository()
+
+            # Test analysis result with correct structure
+            test_insights = [
+                Insight(
```

Ruff (0.12.2)

Undefined name `Insight`

(F821)